config/opal_check_cuda.m4

```diff
@@ -50,9 +50,18 @@ AC_ARG_WITH([cuda],
             [AS_HELP_STRING([--with-cuda(=DIR)],
             [Build cuda support, optionally adding DIR/include])])
 AC_MSG_CHECKING([if --with-cuda is set])
+
+# Search for libcuda.so in $with_cuda if the user didn't pass --with-cuda-libdir
+# Otherwise check for cuda in the default path, /usr/local/cuda. If the default
+# path doesn't exist, set with_cuda_libdir to empty.
```

Comment:
What exactly does it mean for a user to specify `--with-cuda=/something`, but `libcuda.so` is not found under that directory (and therefore you fall back to searching under `/usr/local/cuda`)?

It is unfortunate that Autoconf makes us overload with `$with_FOO` values with `yes`, `no`, or a path.  I.e., there's both *user intent* and *possible directory information* included in the `$with_cuda` variable.

Should you separate out the intent from the directory information first (e.g., make a `WANT_CUDA` variable that is 0 or 1), and leave just the directory -- if there is one -- in `$with_cuda`?